### ISSUE TITLE
SNUtils requires login while running in Firefox Multi-account Container session + SSO auth

### ISSUE THREAD

**Issue by Zorakie** [arnoudkooi/ServiceNow-Utils#415]
Hi,
In Firefox, when you open a new Container session (by default long press on + in the tab bar and select one of the colored containers, which basically creates an isolated session), SNUtils starts to require authentication.

In that session, if you just normally log in, you work with ServiceNow, but if you open SNUtils, no data will be loaded, and it will either show you a red error box with failed authentication, or it will, in some cases, even actively prompt to put in credentials to log in.

There are 2 apparent issues: 
1) SNUtils should inherit the login cookie/token from the running session. Also, I know that SNUtils are not loading the credentials from the Uncolodred (=no container) session, as I have on no-session tab open where SNUtils does work and in parallel a colored container session, where it is not loading anything and asks for the user to log in.
2) Our admin accounts are using SSO, which seems to be not supported, so even if I would gladly put credentials to SNUtils, I can not.

Can this be reviewed, please?

Thank you and have a nice day!

**Comment by arnoudkooi** [comment]
Thanks for raising this, I'm afraid I don't have a quick fix. 
The Bugzilla bug / feature request related to this is: https://bugzilla.mozilla.org/show_bug.cgi?id=1670278

I'll do further investigation to see if there is a workaround and will keep this issue updated.

**Comment by arnoudkooi** [comment]
Can you try version 7.0.6.0 and let me know if this fixes (some) of the issues?

**Comment by Zorakie** [comment]
Hi, thank you for the hotfix. Seems like it could be working.
1) After the update, SNUtils did still prompt for login credentials (which I still cannot put in due to the SSO login).
2) The login prompt changed. Previously, it was a pop-up modal inside of the SNUtils frame. Now, it was an independent movable window/popup outside of the SNUtils frame
3) Surprisingly, when I have canceled it, it still seems to be working. Record data are loading, Nodes are loading...

So in conclusion, except of that false prompt, all good for me 👍

**Comment by arnoudkooi** [comment]
Good, If you do get prompt again let me know.
I added a small fix 7.0.6.1 that opens links from the popup in the correct container as well.

**Comment by Zorakie** [comment]
Hi!
The login prompt I believe it is not fully fixed.
After the last update, I observer the following behavior:
If you open a first brand new session, that you have not opened before, the login prompt will trigger.
If you open 2nd-nth brand new session that you have not opened before, there is no longer a prompt and SNUtils already caught up.
Anonymous mode seems to be surprisingly fine.

Steps that I performed.
1) Browser restarted.
2) Worked in an uncolored session, and my other frequently used "blue" session, no prompts there since your update
3) Randomly I have opened a "green" session, never used for SN. I received a prompt, which I have canceled and everything works, as per the previous report.
4) Out of curiosity, created (did not exist before at all) a new "purple" container session. Loaded SN and SNUtils did NOT prompt for credentials.
5) Disabled SNUtils and enabled SNUtils onPrem.
6) Opened SN in some colored session. Received prompt.
7) Switched to another new colored session. Didn't receive the prompt.

Seems like it is either:
- some cache issue (on my end)
- there is a scenario, where for just the very first time SNUtils loads in any container, it prompts, but then it writes something somewhere so it knows it shouldn't ask again.

**Comment by arnoudkooi** [comment]
I'm not able to reproduce, what is the endpoint URL when you get the prompt?

**Comment by Zorakie** [comment]
A few different ones, I believe one was just a random dashboard after the backend loads and the second one was like sys_user list. All in the old UI16.

so for example, the dashboard was the Admin console and the target link after the full load was
https://<instance>.service-now.com/nav_to.do?uri=%2F$pa_dashboard.do%3Fsysparm_dashboard%3D283eeaa0c730030020508b3c5c9763b3%26sysparm_breakdown_source%3D7e945739c7000300b0ea8b3c5c976319%26sysparm_element%3D%26sysparm_element_value%3D%26sysparm_tab%3D964eaaa0c730030020508b3c5c9763a3

and the second one would be simply like https://<instance>.service-now.com/nav_to.do?uri=%2Fsys_user_list.do


But you know what? If you can not reproduce it, maybe it was my issue/cache. I have 2 more computers at home that also have Firefox and SNUtils, so I will check on those }as I can no longer force the prompt (because I used my containers already) and I will either confirm, or close this topic with this marking as a fluke.

**Comment by arnoudkooi** [comment]
Seems hard to reproduce indeed

btw:
It actually should call a table API endpoint, maybe you could check via developer tools, or it may show after entering (dummy) credentials.

**Comment by Zorakie** [comment]
Hi, I have tested it on my other machine. SNUtils 7.0.8.0 (apparently, there was some other update in the meantime), but everything worked fine. I have opened a new container never used with SN before, and no credential window popped up, so I would call my previous report a fluke and probably just some cache issue or something.

Thus I would like to thank you again for a quick fix and I can close this topic as solved.

**Comment by arnoudkooi** [comment]
Did you try a reinstall?
If you're on SN Devs Slac DM me, maybe we can get on a call to troubleshoot

**Comment by Zorakie** [comment]
Hi, ok, so I can confirm, this issue with false credentials prompt is still present. Just woke up my computer from sleep after the weekend, needed to work in my "blue" session that I normally also use for SN and was using it during last week, and it did indeed show the prompt.

Again, after canceling the prompt, everything works just fine.

**Comment by arnoudkooi** [comment]
Can you do some debugging yourself? Not sure how to solve this as I cant replicate the issue.

**Comment by maialithar** [comment]
I also experience this behaviour (see attachment). You need this plugin to replicate: https://addons.mozilla.org/pl/firefox/addon/multi-account-containers/.
I tried removing SNUtils and reinstalling, but it didn't help.
Firefox 115.02, SNUtils 7.0.8.0

https://github.com/arnoudkooi/ServiceNow-Utils/assets/211896/a51589e2-abae-4a97-9b1b-5a80687b2aa3

**Comment by maialithar** [comment]
the issue seems to have been fixed in 7.0.8.3

**Comment by arnoudkooi** [comment]
Hoping it is solved when you try 7.0.8.3, if not please open a new issue, or we can reopen this one

**Comment by Zorakie** [comment]
Sorry for a late response. After using 7.0.8.1+, I no longer see the issue, even after restarts, etc. so as long as other users don't have still this issue, for now, all good for me. Anyway, thank you very much for the fix!

**Comment by iz0mb** [comment]
I also apologize for the delayed response. I can confirm this has fixed the issue for me as well. Very much appreciate your responsiveness and quick turnaround.
